Dockerfile

```diff
@@ -1,108 +1,159 @@
-FROM alpine:3.12 as bitcoin-core
-
-RUN sed -i 's/http\:\/\/dl-cdn.alpinelinux.org/https\:\/\/alpine.global.ssl.fastly.net/g' /etc/apk/repositories
-RUN apk --no-cache add autoconf
-RUN apk --no-cache add automake
-RUN apk --no-cache add boost-dev
-RUN apk --no-cache add build-base
-RUN apk --no-cache add chrpath
-RUN apk --no-cache add file
-RUN apk --no-cache add gnupg
-RUN apk --no-cache add libevent-dev
-RUN apk --no-cache add libressl
-RUN apk --no-cache add libtool
-RUN apk --no-cache add linux-headers
-RUN apk --no-cache add zeromq-dev
+FROM debian:bullseye-slim as downloader
+
 RUN set -ex \
-    && for key in \
-    90C8019E36C2E964 \
-    ; do \
-    gpg --batch --keyserver keyserver.ubuntu.com --recv-keys "$key" || \
-    gpg --batch --keyserver pgp.mit.edu --recv-keys "$key" || \
-    gpg --batch --keyserver keyserver.pgp.com --recv-keys "$key" || \
-    gpg --batch --keyserver ha.pool.sks-keyservers.net --recv-keys "$key" || \
-    gpg --batch --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys "$key" ; \
-    done
-
-ARG BITCOIN_VERSION
-RUN test -n "$BITCOIN_VERSION"
-
-ENV BITCOIN_PREFIX=/opt/bitcoin-${BITCOIN_VERSION}
-
-RUN wget https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/SHA256SUMS.asc
-RUN wget https://bitcoincore.org/bin/bitcoin-core-${BITCOIN_VERSION}/bitcoin-${BITCOIN_VERSION}.tar.gz
-RUN gpg --verify SHA256SUMS.asc
-RUN grep " bitcoin-${BITCOIN_VERSION}.tar.gz\$" SHA256SUMS.asc | sha256sum -c -
-RUN tar -xzf *.tar.gz
-
-WORKDIR /bitcoin-${BITCOIN_VERSION}
-
-RUN sed -i '/AC_PREREQ/a\AR_FLAGS=cr' src/univalue/configure.ac
-RUN sed -i '/AX_PROG_CC_FOR_BUILD/a\AR_FLAGS=cr' src/secp256k1/configure.ac
-RUN sed -i s:sys/fcntl.h:fcntl.h: src/compat.h
-RUN ./autogen.sh
-RUN ./configure LDFLAGS=-L`ls -d /opt/db*`/lib/ CPPFLAGS=-I`ls -d /opt/db*`/include/ \
-    --prefix=${BITCOIN_PREFIX} \
-    --mandir=/usr/share/man \
-    --disable-tests \
-    --disable-bench \
-    --disable-ccache \
-    --with-gui=no \
-    --disable-wallet \
-    --enable-util-cli
-RUN make -j$(($(nproc) - 1))
-RUN strip ./src/bitcoin-cli
-
-# Builder
-FROM alpine:3.12 as builder
-
-RUN apk add ca-certificates alpine-sdk autoconf automake git libtool gmp-dev \
-    sqlite-dev python2 python3 py3-pip py3-mako net-tools zlib-dev libsodium gettext jq
-RUN pip3 install mrkd mistune==0.8.4
-
-ADD ./.gitmodules /root/.gitmodules
-ADD ./.git /root/.git
```

Comment:
:speech_balloon: Thanks for removing this, this was getting on my sanity.

Reply:
Ah, I learn it still doesn't fix.

Reply:
Yeah, sorry, the build still re-executes every time. I can't figure out how to make it use the docker cache.